bob/bob.go

```diff
@@ -1,15 +1,37 @@
-// This is a "stub" file.  It's a little start on your solution.
-// It's not a complete solution though; you have to write some code.
-
-// Package bob should have a package comment that summarizes what it's about.
-// https://golang.org/doc/effective_go.html#commentary
 package bob
 
-// Hey should have a comment documenting it.
+import (
+	"regexp"
+	"strings"
+)
+
+// Hey function which responds correctly
 func Hey(remark string) string {
-	// Write some code here to pass the test suite.
-	// Then remove all the stock comments.
-	// They're here to help you get started but they only clutter a finished solution.
-	// If you leave them in, reviewers may protest!
-	return ""
-}
+	remark = strings.Trim(remark, " \t\n\r")
+
+	re, _ := regexp.Compile("[a-zA-Z]+")
+
+	// Check if the string was all capitals
+	yelled := strings.ToUpper(remark) == remark && (len(re.FindAllString(remark, -1)) > 0)
+	
+	// Check if the string had a question mark
+	question := strings.HasSuffix(remark, "?")
+
+	if yelled {
+		if question {
+			return "Calm down, I know what I'm doing!"
+		}
+		return "Whoa, chill out!"
+	}
+
+	if question {
+		return "Sure."
+	}
+
+	if remark == "" {
+		return "Fine. Be that way!"
+	}
+
+	return "Whatever."
```

Comment:
Have a look at the `switch` statement (https://github.com/golang/go/wiki/Switch), you will probably find you can tidy up these conditionals a little.